src/site/content/en/blog/building-a-breadcrumbs-component/index.md

```diff
@@ -0,0 +1,498 @@
+
+The browser reads the SVG HTML, puts the icon information into memory, and
+continues with the rest of the page referencing the ID for additional uses of
+the icon, like this:
+
+```html
+<svg viewBox="0 0 24 24" width="24" height="24" aria-hidden="true">
+  <use href="#icon-home" />
+</svg>
+
+<svg viewBox="0 0 24 24" width="24" height="24" aria-hidden="true">
+  <use href="#icon-dropdown-arrow" />
+</svg>
+```
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/BVkOZ3YnQBgF6K09sLTE.png",
+alt="devtools showing a rendered `<use>` SVG element", width="800", height="447"
+%}
+
+Define once, use as many times as you like, with minimal page performance impact
+and flexible styling. Notice `aria-hidden="true"` is added to the SVG element.
+The icons are not useful to someone browsing who only hears the content, hiding
+them from those users stops them adding unnecessary noise.
+
+#### Split-link `.crumb`
+
+This is where the traditional breadcrumb and those in this component diverge.
+Normally, this would only be an `<a>` link, but I've added traversal UX with a
+disguised select. The `.crumb` class is responsible for laying out the link and
+the icon, while the `.crumbicon` is responsible for stacking the icon and select
+element together. I've called it a split-link because its functions are very
+similar to a [split-button](https://www.nngroup.com/articles/split-buttons/),
+but for page navigation.
+
+```html
+<span class="crumb">
+  <a href="#sub-collection-b">Category B</a>
+  <span class="crumbicon">
+    <svg>...</svg>
+    <select class="disguised-select" title="Navigate to another category">
+      <option>Category A</option>
+      <option selected>Category B</option>
+      <option>Category C</option>
+    </select>
+  </span>
+</span>
+```
+
+A link and some options is nothing special but adds more functionality to a
+simple breadcrumb. Adding a `title` to the select element is helpful for screen
+reader users, giving them information about the action of the button. However it
+provides the same help to everyone else too, you'll see it's front and center on
+iPad. One attribute provides button context to many users. 
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/IbANGFRfyEC61P7ETlGp.png",
+alt="screenshot with the invisible select element being hovered and its
+contextual tooltip showing", width="271", height="83" %}
+
+#### Separator decorations
+
+```html
+<span class="crumb-separator" aria-hidden="true">→</span>
+```
+
+{% Video
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/LASjAldbKTMivtF01aWo.mp4",
+  class="w-screenshot",
+  autoplay="true",
+  loop="true",
+  muted="true"
+%}
+
+Thanks to the [`gap`](https://web.dev/flexbox-gap/) property, a feature for
+optional separators was easy to implement. Not only are separators optional
+entirely, adding just one works great too (see the 3rd example in the video
+above). I then give each `aria-hidden="true"` since they are decorative and not
+something a screen reader needs to announce.
```

Comment:
```suggestion
Separators are optional, adding just one works great too (see the 3rd example in the video
above). I then give each `aria-hidden="true"` since they are decorative and not
something a screen reader needs to announce.

The [`gap`](/flexbox-gap) property, covered next, makes the spacing of these straightforward.
```

Reply:
y'all help these articles so much 😍